File: Tareas/proyecto/app/app/src/main/java/com/example/proyecto/MainActivity.java
package com.example.proyecto;


import androidx.appcompat.app.AppCompatActivity;
import androidx.core.app.ActivityCompat;

import android.Manifest;
import android.annotation.SuppressLint;
import android.bluetooth.BluetoothAdapter;
import android.bluetooth.BluetoothDevice;
import android.bluetooth.BluetoothSocket;
import android.content.Intent;
import android.content.pm.PackageManager;
import android.os.Bundle;
import android.os.Handler;
import android.view.MotionEvent;
import android.view.View;
import android.widget.Button;
import android.widget.ListView;
import android.widget.Toast;

import java.io.IOException;
import java.io.InputStream;
import java.io.OutputStream;
import java.util.Set;
import java.util.UUID;

public class MainActivity extends AppCompatActivity {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        ListView dispLista = findViewById(R.id.listD);

        final BluetoothJhr bluetoothJhr = new BluetoothJhr(this, dispLista, Main2Activity.class);

        bluetoothJhr.onBluetooth();

        dispLista.setOnItemClickListener((adapterView, view, i, l) -> {
            bluetoothJhr.bluetoothSeleccion(i);
        });
    }

}

File: Tareas/proyecto/kotlin/app/src/main/java/com/example/myapplication/MainActivity.kt
package com.example.myapplication

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*


class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)


        val blue = BluetoothJhr(this,disp, Main2Activity::class.java)
        //para listView normal solo accedemos a onBluetooth
        blue.onBluetooth()

        //para listView Custom return ArrayList<String>
        //val a = blue.dispEmparejados()

        //blue.bluetoothSeleccionAddres("Addres")

        disp.setOnItemClickListener { adapterView, view, i, l ->
            blue.bluetoothSeleccion(i)
        }
    }
}